File: Java/RecyclerView5RecyclerViewInsideRecyclerView/app/src/main/java/dev/ogabek/recyclerview5_recyclerviewinsiderecyclerview/activity/MainActivity.java
package dev.ogabek.recyclerview5_recyclerviewinsiderecyclerview.activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Context;
import android.os.Bundle;

import java.util.ArrayList;
import java.util.List;

import dev.ogabek.recyclerview5_recyclerviewinsiderecyclerview.R;
import dev.ogabek.recyclerview5_recyclerviewinsiderecyclerview.adapter.MainRecyclerViewAdapter;
import dev.ogabek.recyclerview5_recyclerviewinsiderecyclerview.model.Member;
import dev.ogabek.recyclerview5_recyclerviewinsiderecyclerview.model.MemberSub;

public class MainActivity extends AppCompatActivity {

    Context context;
    RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();

        List<Member> members = prepareMembers();
        refreshAdapter(members);

    }

    private void refreshAdapter(List<Member> members) {
        MainRecyclerViewAdapter adapter = new MainRecyclerViewAdapter(context, members);
        recyclerView.setAdapter(adapter);
    }

    void initViews() {
        recyclerView = findViewById(R.id.rv_mainRecyclerView);
        recyclerView.setLayoutManager(new GridLayoutManager(context, 1));
    }

    private List<Member> prepareMembers() {
        List<Member> members = new ArrayList<>();
        for (int i = 1; i < 15; i++) {
            if (i % 4 == 0) {
                members.add(new Member("Ogabek Matyakubov", "+998 99 749 25 81", prepareSubMembers()));
            } else {
                members.add(new Member("Ogabek Matyakubov", "+998 99 749 25 81", null));
            }
        }
        return members;
    }

    private List<MemberSub> prepareSubMembers() {
        List<MemberSub> memberSubs = new ArrayList<>();
        for (int i = 0; i < 10; i++) {
            memberSubs.add(new MemberSub());
        }
        return memberSubs;
    }

}

File: Kotlin/RecyclerView5RecyclerViewInsideRecyclerView/app/src/main/java/dev/ogabek/recyclerview5_recyclerviewinsiderecyclerview/activity/MainActivity.kt
package dev.ogabek.recyclerview5_recyclerviewinsiderecyclerview.activity

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import dev.ogabek.recyclerview5_recyclerviewinsiderecyclerview.R
import dev.ogabek.recyclerview5_recyclerviewinsiderecyclerview.adapter.MainAdapter
import dev.ogabek.recyclerview5_recyclerviewinsiderecyclerview.model.Member

class MainActivity : AppCompatActivity() {

    private lateinit var context: MainActivity
    private lateinit var recyclerView: RecyclerView
    private lateinit var adapter: MainAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initViews()

        refreshAdapter(prepareMemberList())

    }

    private fun initViews() {
        context = this
        recyclerView = findViewById(R.id.rv_main_recycler_view)
        recyclerView.layoutManager = GridLayoutManager(context, 1)

    }

    private fun refreshAdapter(members: List<Member>) {
        adapter = MainAdapter(context, members)
        recyclerView.adapter = adapter
    }

    private fun prepareMemberList(): List<Member> {
        val members = ArrayList<Member>()
        for (i in 0..50) {
            members.add(Member(R.drawable.img_user, "Ogabek Matyakubov", "Mobile Developer"))
        }
        return members
    }
}